java-checks/src/main/java/org/sonar/java/checks/CollectionImplementationReferencedCheck.java

```diff
@@ -177,4 +330,44 @@ private static String toSimpleName(String fullyQualifiedName) {
   private static boolean isPublic(ModifiersTree modifiers) {
     return ModifiersUtils.hasModifier(modifiers, Modifier.PUBLIC);
   }
+
+  private static <T> Set<T> union(Set<T> a, Set<T> b) {
```

Comment:
Since this method is only used to join set of strings, we can probably drop the type parameter and just use `String` as the type. WDYT?

Reply:
Replaced it with `SetUtils.concat` from `analyzer-commons`.